federation/pkg/federation-controller/dns/dns_test.go

```diff
@@ -14,74 +14,154 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package service
+package dns
 
 import (
-	"sync"
 	"testing"
 
 	"fmt"
 	"reflect"
 	"sort"
 
-	"k8s.io/kubernetes/federation/apis/federation/v1beta1"
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns" // Only for unit testing purposes.
-	"k8s.io/kubernetes/pkg/api/v1"
-	"k8s.io/kubernetes/pkg/util/sets"
+	sc "k8s.io/kubernetes/federation/pkg/federation-controller/service"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
 )
 
 func TestServiceController_ensureDnsRecords(t *testing.T) {
 	tests := []struct {
-		name          string
-		service       v1.Service
-		expected      []string
-		serviceStatus v1.LoadBalancerStatus
+		name     string
+		service  apiv1.Service
+		exist    bool
+		expected []string
 	}{
 		{
-			name: "withip",
-			service: v1.Service{
-				ObjectMeta: v1.ObjectMeta{
-					Name:      "servicename",
-					Namespace: "servicenamespace",
+			name: "ServiceWithSingleIpEndpoint",
+			service: apiv1.Service{
+				ObjectMeta: apiv1.ObjectMeta{
+					Name:      "nginx",
+					Namespace: "ns",
+					Annotations: sc.NewEndpointAnnotation(sc.NewEpMap().
+						AddEndpoint("fooregion", "foozone", "c1", []string{"198.51.100.1"}, true).
+						GetJSONMarshalledBytes()),
 				},
 			},
-			serviceStatus: buildServiceStatus([][]string{{"198.51.100.1", ""}}),
+			exist: true,
 			expected: []string{
-				"example.com:servicename.servicenamespace.myfederation.svc.federation.example.com:A:180:[198.51.100.1]",
-				"example.com:servicename.servicenamespace.myfederation.svc.fooregion.federation.example.com:A:180:[198.51.100.1]",
-				"example.com:servicename.servicenamespace.myfederation.svc.foozone.fooregion.federation.example.com:A:180:[198.51.100.1]",
+				"example.com:nginx.ns.myfed.svc.fed.example.com:A:180:[198.51.100.1]",
```

Comment:
It's not clear what the purpose of these changes is?  You've changed from generic "servicename" to "nginx".  Why?